src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -1040,6 +1040,7 @@ winrt::fire_and_forget Pane::_ControlConnectionStateChangedHandler(const winrt::
         newConnectionState = coreState.ConnectionState();
     }
 
+    const auto previousConnectionState = std::exchange(_connectionState, newConnectionState);

```

Comment:
/cc @lhecker to make sure I didn't royally blow up some locking / co_await / other threading issue, which I am prone to do

Reply:
Not that I can tell at least!